### PR TITLE
#167484603 Fetch all available profiles

### PR DESCRIPTION
**Pull request by malaba6**
### What does this PR do?
- Creates the endpoint to fetch all the user profiles
- Tests the endpoints
- Adds the user's endpoint in the API documentation
- Fixes the `user can successfully login test`

### Description of Task to be completed?
- Create a method called `fetchProfiles` inside the class `Profile` in the `profileController` file.
- Add the route in the `routes` folder to fetch profiles with the `get` method
- In the `userProfile.test` file, add tests for this endpoint
- In `swagger.json` file, add `/api/v1/profiles` in the swagger file's path to document this endpoint

### How should this be manually tested?
In Postman or Swagger or any other tool of your choice visit this enpoint:
`/api/v1/profiles` to fetch a list of profiles

### Any background context you want to provide:
To access this endpoint, User must be authenticated. And use the token generated for Loggin or Registration as `Authorisation` in the 'header'.

### What are the relevant pivotal tracker stories?
[List user functionality](https://www.pivotaltracker.com/story/show/167484603)

### Screenshots? 
*In Postman*
Screen Shot 2019-08-14 at 04 22 29

*In swagger*
Screen Shot 2019-08-14 at 04 20 36

### CODE REVIEW

File: src/controllers/profileController.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    }
+  }
+
+  /**
+       * @description update user profile
+       * @param {object} req
+       * @param {object} res
+       * @returns {object} return object containing updated user profile
+       */
+  static async editProfile(req, res) {
```
Comment (harerakalex) [inline src/controllers/profileController.js:71]: According to the title of this PR, `#167484603 Fetch all available profiles`, and then this method is to edit profile. Why did you create it?

**Review by raymond42 — CHANGES_REQUESTED** [review]
Working good. But first, there are some conflicts needed to be resolved

File: src/controllers/profileController.js (hunk 1 above)
Comment (malaba6) [reply src/controllers/profileController.js:71]: I did not create that method. I just edited the class where that method was created, Raymond created it

**Comment by malaba6** [comment]
> Working good. But first, there are some conflicts needed to be resolved

On it. This happened because of the merging. Let me rebase and get rid of the conflict

File: src/controllers/profileController.js [hunk 2]
```diff
@@ -42,11 +42,32 @@ class Profile {
...
+      const users = await User.findAll({
+        where: { role: 'user' },
+        attributes: ['userName', 'bio', 'image']
+      });
+      res.status(200).json({
+        message: `${users.length} Users found`,
+        data: users
+      });
+    } catch (err) {
+      // console.log(err);
```
Comment (mkiterian) [inline src/controllers/profileController.js]: do not mute the error. I'd rather you throw the error, this is important for debugging purposes

File: src/tests/userprofile.test.js [hunk 3]
```diff
@@ -40,6 +42,68 @@ describe('GET /api/v1/profiles', () => {
...
+        expect(res.body).to.be.an('object');
+        expect(res.body).to.have.keys('error');
+        expect(res.body.error).to.deep.equal('Please log in or Register');
+        done();
+      });
+  });
+});
+
+describe('GET /api/v1/profiles', () => {
+  it('Should A list of profiles if  user is identified', (done) => {
```
Comment (mkiterian) [inline src/tests/userprofile.test.js]: grammar: "Should return a list ..."

File: src/tests/userprofile.test.js [hunk 4]
```diff
@@ -40,6 +42,68 @@ describe('GET /api/v1/profiles', () => {
...
+        expect(res).to.have.status(401);
+        expect(res.body).to.be.an('object');
+        expect(res.body).to.have.keys('error');
+        expect(res.body.error).to.deep.equal('Please log in or Register');
+        done();
+      });
+  });
+});
+
+describe('GET /api/v1/profiles', () => {
```
Comment (mkiterian) [inline src/tests/userprofile.test.js]: this looks like a copy of https://github.com/andela/coding-geeks-ah-backend/pull/19/files#diff-7e80fd915c1e3097f4adf33a8a3c7f0dR60

Comment (mkiterian) [reply src/tests/userprofile.test.js]: I'd also like to see an assertion about the number of profiles returned is what we expect

**Review by mkiterian — CHANGES_REQUESTED** [review]
I also think the `/api/profiles` endpoint shouldn't require authentication. You can also remove the auth requirement on `/api/profiles/:username`

**Comment by malaba6** [comment]
> I also think the `/api/profiles` endpoint shouldn't require authentication. You can also remove the auth requirement on `/api/profiles/:username`

I authenticated this endpoint because it is required according to the story description on the PT board here 
Screen Shot 2019-08-15 at 19 57 11

for `/api/profiles/:username`, I removed the authentication before I pushed, so it does not require authentication

File: src/controllers/profileController.js (hunk 2 above)
Comment (malaba6) [reply src/controllers/profileController.js]: Okay...fixed!

File: src/tests/userprofile.test.js (hunk 3 above)
Comment (malaba6) [reply src/tests/userprofile.test.js]: Alright! done!

File: src/tests/userprofile.test.js (hunk 4 above)
Comment (malaba6) [reply src/tests/userprofile.test.js]: > this looks like a copy of https://github.com/andela/coding-geeks-ah-backend/pull/19/files#diff-7e80fd915c1e3097f4adf33a8a3c7f0dR60

Ok...fixed

Comment (malaba6) [reply src/tests/userprofile.test.js]: > I'd also like to see an assertion about the number of profiles returned is what we expect

On it...

**Pull request merged**